test/configs/ServerConfig.ts

```diff
@@ -0,0 +1,5 @@
+import { HttpHandler } from '../../src/server/HttpHandler';
+
+export interface ServerConfig {
```

Comment:
Haven't gone through this PR in detail, but wouldn't it be easier to define configs in JSON? Like `config/config-default.json`.
That way, we'd immediately also be testing the DI layer.

Reply:
The idea was to have this testing part be independent of the dependency injection layer. We could have separate system tests to test that later though.

Reply:
Ok, fair enough.